model/src/order.rs

```diff
@@ -113,7 +110,8 @@ impl OrderBuilder {
     pub fn sign_with(mut self, domain_separator: &DomainSeparator, key: SecretKeyRef) -> Self {
         self.0.order_meta_data.owner = key.address();
         self.0.order_meta_data.uid = self.0.order_creation.uid(&key.address());
-        self.0.order_creation.sign_self_with(domain_separator, &key);
+        self.0.order_creation.signature =
+            self.0.order_creation.sign_self_with(domain_separator, &key);
```

Comment:
This was the Bug from the previous PR. I hadn't changed this line which was causing the `OrderBuilder.sign_with` to be incorrect